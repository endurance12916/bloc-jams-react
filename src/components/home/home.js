import React, { Component } from 'react';
import './home.css';

const DefaultText = () => (
    <section className="hero-content"> 
        <h1 className="hero-title">Turn the music up!</h1>
    </section>
)

class ScrollText extends Component {

    animatePoints(points) {
        for (let point of points) {
            point.style.opacity = 1;
            point.style.transform = "scaleX(1) translateY(0)";
            point.style.msTransform = "scaleX(1) translateY(0)";
            point.style.WebkitTransform = "scaleX(1) translateY(0)";
        }
    };

    scrollDown = (event) => {
        // const pointsArray = document.getElementsByClassName('point');
        const pointsArray = [this.pointOne, this.pointTwo, this.pointThree]
        const scrollDistance = this.sellingPoints.getBoundingClientRect().top - window.innerHeight + 200;
        // Automatically animate the points on a tall screen where scrolling can't trigger the animation
        if (window.innerHeight > 950) {
            this.animatePoints(pointsArray);
        }
        //  console.log("Current offset from the top is " + sellingPoints.getBoundingClientRect().top + " pixels");
        if (document.documentElement.scrollTop || document.body.scrollTop >= scrollDistance) {
            this.animatePoints(pointsArray);
        }
    }

    componentDidMount() {
        window.addEventListener("scroll", this.scrollDown);
    }

    componentWillUnmount() {
        window.removeEventListener("scroll", this.scrollDown)
    }

    render(){
        return (
        <section className="selling-points container clearfix" ref={(sellingPoints) => { this.sellingPoints = sellingPoints }} >
            <div className="point column third" ref={(pointOne) => { this.pointOne = pointOne }}>
                <span className="ion-music-note"></span>
                <h5 className="point-title">Choose your music</h5>
                <p className="point-description">The world is full of music; why should you have to listen to music that someone else chose?</p>
            </div>
            <div className="point column third" ref={(pointTwo) => { this.pointTwo = pointTwo }}>
                <span className="ion-radio-waves"></span>
                <h5 className="point-title">Unlimited, streaming, ad-free</h5>
                <p className="point-description">No arbitrary limits. No distractions.</p>
            </div>
            <div className="point column third" ref={(pointThree) => { this.pointThree = pointThree }}>
                <span className="ion-iphone"></span>
                <h5 className="point-title">Mobile enabled</h5>
                <p className="point-description">Listen to your music on the go. This streaming service is available on all mobile platforms.</p>
            </div>
        </section>
        )
    }
}

const Home = () => {
    return (
        <section className="landing">
            <DefaultText />
            <ScrollText />
        </section>
    );
}


// export default Home;
export { Home, DefaultText, ScrollText }