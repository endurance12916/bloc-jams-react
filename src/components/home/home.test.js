import { Home, DefaultText, ScrollText } from './home'
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  mount(<Home />)
});

it('renders a snapshot', () => {
    const wrapper = renderer.create(<Home />).toJSON();
    expect(wrapper).toMatchSnapshot();
});

// Why does this one return error, even though the first test (renders without crashing) passes?
it('renders default text', () => {
  const wrapper = mount(<Home />)
  const expectedNode = shallow(<DefaultText />)
  expect((wrapper).contains(expectedNode)).toEqual(true);
})

// This test doesn't work...How to test a scrolling event?
it('renders additional texts after scrolling', () => {
  const scrolling = mount(<Home />).simulate('scroll');
  expect(scrolling.contains(ScrollText)).toEqual(true);
})