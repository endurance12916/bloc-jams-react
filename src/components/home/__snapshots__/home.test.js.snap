exports[`test renders a snapshot 1`] = `
<section
  className="landing">
  <section
    className="hero-content">
    <h1
      className="hero-title">
      Turn the music up!
    </h1>
  </section>
  <section
    className="selling-points container clearfix">
    <div
      className="point column third">
      <span
        className="ion-music-note" />
      <h5
        className="point-title">
        Choose your music
      </h5>
      <p
        className="point-description">
        The world is full of music; why should you have to listen to music that someone else chose?
      </p>
    </div>
    <div
      className="point column third">
      <span
        className="ion-radio-waves" />
      <h5
        className="point-title">
        Unlimited, streaming, ad-free
      </h5>
      <p
        className="point-description">
        No arbitrary limits. No distractions.
      </p>
    </div>
    <div
      className="point column third">
      <span
        className="ion-iphone" />
      <h5
        className="point-title">
        Mobile enabled
      </h5>
      <p
        className="point-description">
        Listen to your music on the go. This streaming service is available on all mobile platforms.
      </p>
    </div>
  </section>
</section>
`;
